### PR TITLE
Directed Tests - `secureseed`

### PR DESCRIPTION
**Pull request by silabs-robin**
This PR adds two directed tests for `secureseed`, one using `rs1=x0` and for for LFSR lockup.

### CODE REVIEW

**Review by silabs-hfegran — COMMENTED** [review]
Looks good to me, waiting for @silabs-krdosvik for final approval

File: cv32e40s/tests/programs/custom/cov_holes_generic/cov_holes_generic.c
Comment (silabs-hfegran) [inline cv32e40s/tests/programs/custom/cov_holes_generic/cov_holes_generic.c:482]: The added tests do not use the generic test-template structure, and does not correspond to the comments in the template. This also causes the standard reporting mechanism not to work as intended. Additionally, there is no way to disable the immediate test-termination on error.

**Review by silabs-hfegran — CHANGES_REQUESTED** [review]

File: cv32e40s/tests/programs/custom/cov_holes_generic/cov_holes_generic.c
Comment (silabs-krdosvik) [reply cv32e40s/tests/programs/custom/cov_holes_generic/cov_holes_generic.c:598]: Looks good. 

Is it possible to detect if the minor alert gets set too? That would be interesting I think.

Since we are making a test for writing 0 to the secureseed registers, I think it would be nice if we also made a test for writing a "bad" seed to the secureseed registers. With "bad" seed I mean that eventually the LFSR would go into lockup by itself. This is expected behavior, and there is rtl code designed to make sure we handle this properly, so yeah it would be nice to test it too, if time allows.

File: cv32e40s/tests/programs/custom/cov_holes_generic/cov_holes_generic.c
Comment (silabs-krdosvik) [reply cv32e40s/tests/programs/custom/cov_holes_generic/cov_holes_generic.c:651]: Looks good!

File: cv32e40s/tests/programs/custom/cov_holes_generic/cov_holes_generic.c
Comment (silabs-robin) [reply cv32e40s/tests/programs/custom/cov_holes_generic/cov_holes_generic.c]: One moment...

File: cv32e40s/tests/programs/custom/cov_holes_generic/cov_holes_generic.c
Comment (silabs-robin) [reply cv32e40s/tests/programs/custom/cov_holes_generic/cov_holes_generic.c]: I realized we already have a test for xsecure csrs (`xsecure_csrs`) so I moved it there.

File: cv32e40s/tests/programs/custom/cov_holes_generic/cov_holes_generic.c
Comment (silabs-krdosvik) [reply cv32e40s/tests/programs/custom/cov_holes_generic/cov_holes_generic.c]: That is Perfect! 
Have you though about my comment about minor alert?

Comment (silabs-krdosvik) [reply cv32e40s/tests/programs/custom/cov_holes_generic/cov_holes_generic.c]: Perfect!

Comment (silabs-krdosvik) [reply cv32e40s/tests/programs/custom/cov_holes_generic/cov_holes_generic.c]: RESOLVED

**Review by silabs-hfegran — APPROVED** [review]
Looks good to me, thanks!

**Pull request merged**
